### PR TITLE
WT-13099 Document locking sequence for eviction

### PR DESCRIPTION
**Pull request by ershov**

### CODE REVIEW

File: src/docs/eviction.dox [hunk 1]
```diff
@@ -25,11 +25,34 @@ Eviction has to obtain exclusive access to the page as there may be
...
+- \c queue->evict_lock
+
+Locking order:
+
+- Eviction server:
+  - lock \c cache->evict_pass_lock
+    - trylock \c cache->evict_walk_lock
+      - lock \c cache->evict_queue_lock
+        - lock \c queue->evict_lock
+- Critical path:
```
Comment (quchenhao) [inline src/docs/eviction.dox:42]: What is the critical path?

File: src/docs/eviction.dox [hunk 2]
```diff
@@ -25,11 +25,34 @@ Eviction has to obtain exclusive access to the page as there may be
 application threads still reading content on the page that is being
 evicted.
 
+Internally eviction uses the following locks:
```
Comment (quchenhao) [inline src/docs/eviction.dox:28]: Can you describe what shared memory each lock protects?

File: src/docs/eviction.dox (hunk 1 above)
Comment (ershov) [reply src/docs/eviction.dox:42]: Re-worded the description.

File: src/docs/eviction.dox (hunk 2 above)
Comment (ershov) [reply src/docs/eviction.dox:28]: Done

**Review by quchenhao — APPROVED** [review]
lgtm

**Pull request merged**